Hosp data

mrajeev08 opened the pull request:
Attempting #58.

I added a function `get_hospdata_wide` which formats in a similar way to other datasets, but I realized that it's a bit tricky given the dual data sources and may not actually be very useful downstream? It also adds `janitor` as a dependency which seems uneccessary too:) So just let me know and I can ditch that.

Also don't know if the carry_fwd_value bits are needed downstream? Tried to explain in the documentation. 

I also attempted to add some tests, but let me know if they don't make sense!

beansrowning commented on R/get_hospdata.R:
```diff
@@ -0,0 +1,159 @@
+    filter(source %in% preferred_source) %>%
+    select(-carry_fwd_value) %>%
+    tidyr::pivot_wider(names_from = indicator, values_from = value) %>%
+    janitor::clean_names()
```
Hm. I might suggest explicitly renaming the columns instead of using `clean_names`, just so we don't have to import an entire package just for that function. I do love janitor, though!

beansrowning commented on R/get_hospdata.R:
```diff
@@ -0,0 +1,159 @@
+    filter(source %in% preferred_source) %>%
+    select(-carry_fwd_value) %>%
+    tidyr::pivot_wider(names_from = indicator, values_from = value) %>%
+    janitor::clean_names()
```
Hm. I might suggest explicitly renaming the columns instead of using `clean_names`, just so we don't have to import an entire package just for that function. I do love janitor, though!

beansrowning replied:
Easy approach: write a lookup in lookups.R that maps raw to clean column names, and use dplyr::recode() in a mutate block above the pivot_wider(). 

I did something similar here: https://github.com/CDCgov/SaviR/blob/640c554e049161fefe8b7f5be814048bb8a604e7/R/get_covid_sources.R#L29

We could also write a test in the data_accessors to check that the column names are present as we expect, so we can see if that structure changes.

beansrowning replied:
Easy approach: write a lookup in lookups.R that maps raw to clean column names, and use dplyr::recode() in a mutate block above the pivot_wider(). 

I did something similar here: https://github.com/CDCgov/SaviR/blob/640c554e049161fefe8b7f5be814048bb8a604e7/R/get_covid_sources.R#L29

We could also write a test in the data_accessors to check that the column names are present as we expect, so we can see if that structure changes.

mrajeev08 replied:
Oh cool, yeah can try that! You think it's worth keeping 'get_hospdata_wide'? I don't know if the wide format ever gets used down stream at the moment...I think the DUB would just use get_hospdata_long -> get_hospdata_latest. Also I don't quite know why the github actions checks are failing (something to do with a vignette but I didn't think I changed anything in there?)!

beansrowning replied:
See below RE: wide and general API stuff.

Not sure on the testing! I'm looking into it. May be upstream/unrelated.

beansrowning replied:
@mrajeev08 FYI, see #62. Looks like a {gt} error.

mrajeev08 commented:
Kay, went with a lazier approach to the names to avoid adding janitor, but also added a test to make sure that indicators get named correctly and match the current versions for the two test datasets.

Still can ditch get_hospdata_wide if you think not of use, however!

I'm using tidyr::pivot_wider, but tidyr isn't in the description file (I think it's used a couple of other places in the pkg though?). Don't know how it works re is it unnecessary to add it to the description file if you call it like pkg::fun in the code?

Let me know what you think (and if checks don't pass if there's something I need to fix:)!

beansrowning commented on R/get_hospdata.R:
```diff
@@ -0,0 +1,159 @@
+#'   hospdata_latest <- get_hospdata_latest(hospdata_long, preferred_source = "OWID")
+#'   head(hospdata_latest)
+#' }
+get_hospdata_latest <- function(hospdata_long,
```
What are your thoughts on this API?

I started thinking about it more in response to wide, and wondered if there was any need for anything other than just a "long" version.

The testing accessors are kind of similar to this I guess, but `get_vax()`, `get_covid_df()`, etc. are all "long" by default. Is there code that relies on the `get_hospdata_latest()`-type of transformation already, or do you think we could just keep only `get_hospdata()` and let the user decide if they want only the latest data or not?

mrajeev08 replied:
I think just [here](https://github.com/CDCgov/ITF_Power_BI/blob/aebe03053d4814478075bee601e2c984f0836814/itf_dashboard/0_output_data.R#L258) and in the DUB it pulls in the file that's generated there. I think looking at the OG functions maybe the rationale was to avoid having to complete the data by date and just filtering to the most recent first. But I thought that would be harder to do in the pkg and since the longest part is just downloading all the data, didn't seem necessary to keep the download and combine separate from the complete step? 

How about I ditch get_hospdata_latest and get_hospdata_wide and instead add a couple of lines showing how to get those outputs to the example in get_hospdata_long? I can also just change that function then just to 'get_hospdata' :) 

Let me know!

beansrowning replied:
I like that idea.

beansrowning commented:
Closes #58

mrajeev08 commented:
Alrighty, changing to single function!
- also added test for indicator source combos
- moved getting recent / wide format to the examples
- added some info on which indicators present in which source in the docs

beansrowning reviewed (approved):
Looks good! Ready to merge

mrajeev08 commented on R/get_hospdata.R:
```diff
@@ -0,0 +1,104 @@
+#' @return A data frame with n rows and 5 columns:
+#'
+#' \describe{
+#'       \item{id}{character, ISO 3166-1 alpha-3 country code}
```
oh cool, good to know! It wasn't bad with some datapasta + ctrl alts:)

mrajeev08 commented:
Shall I go ahead and merge & close even though that one check is failing (was that the openssl thingy)?

beansrowning commented:
Yeah, let's ignore for now. Linux release and dev show fine.

mrajeev08 commented:
Closes #58
